README.md

```diff
@@ -1,26 +1,32 @@
 # umple.gradle.plugin
 
 **Tasks:**
-  - compileUmpleFile
+  - generateSource
 
 **Instructions:**
-To use, you must set the relevant project properties:
-  - umpleFileName (**required**)
-  - languageToGenerate (**required**)
-  - outputPath (**optional**)
+To use the plugin, you set the relevant plugin properties:
+  - umpleFilePath
+  - languageToGenerate
+  - outputPath
 
-These can be set either in a script, or through the command line. If any required fields are not set, an exception will be thrown.
-  
-**Script:**
-```
-project.ext.set("umpleFileName", "test.ump")
-project.ext.set("languageToGenerate", "Java")
-```
+These values are set in the build.gradle file of the project you're applyung the plugin to. If any required fields are not set, the plugin will use the following default values:
+
+languageToGenerate = 'Java' 
+outputPath = 'generated/java'
+umpleFilePath = 'src/umple/master.ump'
 
-**Command Line**
+The plugin properties are attached to a SourceSet called generatedSource. You can declare the source set and set the plugin properties on the source set by including the following snippet in your project's build.gradle file:
+  
 ```
-gradle compileUmpleFile -PumpleFileName=test.ump -PlanguageToGenerate=Java
+sourceSets {
+    generatedSource { 
+       ext.languageToGenerate = 'Java' 
```

Comment:
I think using source sets is still a good idea. For one thing, it allows for use of the `compileGeneratedSourceJava` task if the generated files are Java. I just wish there was a way to make it a bit more light-weight.